central/graphql/resolvers/node_components.go

```diff
@@ -246,7 +304,22 @@ func (resolver *nodeComponentResolver) NodeCount(ctx context.Context, args RawQu
 // NodeVulnerabilities contained in the node component
 func (resolver *nodeComponentResolver) NodeVulnerabilities(ctx context.Context, args PaginatedQuery) ([]NodeVulnerabilityResolver, error) {
 	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.NodeComponents, "NodeVulnerabilities")
-	return resolver.root.NodeVulnerabilities(resolver.nodeComponentScopeContext(ctx), args)
+
+	if resolver.ctx == nil {
+		resolver.ctx = ctx
+	}
+
+	// Short path. Full node is embedded when node scan resolver is called.
+	embeddedComponent := embeddedobjs.NodeComponentFromContext(resolver.ctx)
+	if embeddedComponent == nil || !env.PostgresDatastoreEnabled.BooleanSetting() {
+		return resolver.root.NodeVulnerabilities(resolver.nodeComponentScopeContext(ctx), args)
```

Comment:
Should this line be using `ctx` or `resolver.ctx`?

Reply:
No it should still use ctx because `resolver.nodeComponentScopeContext` func also does below assignment
```
if resolver.ctx == nil {
 		resolver.ctx = ctx
 	}
```
It is redundant in this case, but we need non-nil resolver.ctx when calling `getNodeCVEResolvers` on ln 322 below. resolver.nodeComponentScopeContext get called only within the if condition, so it won't be always executed and assign resolver.ctx to a non-nil value.